static/src/javascripts/projects/common/modules/experiments/tests/contributions-epic-thank-you.js

```diff
@@ -0,0 +1,169 @@
+                $('.contributions__epic').each(function (el) {
+                        //top offset of 18 ensures view only counts when half of element is on screen
+                        var elementInview = ElementInview(el, window, {top: 18});
+                        elementInview.on('firstview', function () {
+                            setValue('gu.thankyouCount', thankYouCounter + 1);
+                        });
+
+                });
+
+            });
+        }
+
+        var makeUrl = function(urlPrefix, intcmp) {
+            return urlPrefix + 'INTCMP=' + intcmp;
+        };
+
+        var membershipUrl = 'https://membership.theguardian.com/supporter?';
+        var contributeUrl = 'https://contribute.theguardian.com/?';
+        var thankyouCount = getValue('gu.thankyouCount') || 0;
+        var campaignCode = 'epic_thankyou_' + thankyouCount;
+
+
+        var link = isPayingMember? makeUrl(contributeUrl, campaignCode) : makeUrl(membershipUrl, campaignCode);
+
+        var cta = {
+            v1: {
+                title: 'We\'d like to say thanks &#8230;',
+                p1: 'Your crucial financial support makes our journalism possible. We do it because we believe, like you, that the world has never needed fearless, independent media more.',
+                p2: 'If you know someone who might want to support us, why not tell them how they can by <a target=\"_blank\" href=\"' + link + '\">sharing this link?</a>'
+            },
+
+            v2: {
+                title: 'Thank you for your support',
+                p1: 'Your crucial financial support makes our journalism possible. We do it because we believe, like you, that the world has never needed fearless, independent media more.',
+                p2: 'If you know someone who might share that perspective and want to support us, why not tell them how they can by <a target=\"_blank\" href=\"' + link + '\">sharing this link?</a>'
+
+            }
+
+        };
+
+
+        this.variants = [
+            {
+                id: 'control',
+
+                test: function () {
+                    addInviewLIstener(thankyouCount);
+                    if(thankyouCount < 4) {
+                        var messages = cta.v1;
+                        var component = $.create(template(contributionsEpicThankYou, {
+                            title: messages.title,
+                            p1: messages.p1,
+                            p2: messages.p2
+                        }));
+                        componentWriter(component);
+                    }
+                },
+
+                impression: function(track) {
+                    mediator.on('contributions-embed:insert', track);
+                },
+
+                success: completer
+            },
+
+            {
+                id: 'other',
+
+                test: function () {
+                    addInviewLIstener(thankyouCount);
+                    if(thankyouCount < 4) {
```

Comment:
Could this be abstracted out to a function that takes `messages` as an argument ?